### PR TITLE
feat: upgrade to docusaurus v3

### PR DESCRIPTION
**Pull request by shimoncohen**
Upgrade to docusaurus v3 using the following guide:
https://docusaurus.io/docs/migration/v3

### CODE REVIEW

**Comment by shimoncohen** [comment]
Considering adding visual regression checks:
https://docusaurus.io/blog/upgrading-frontend-dependencies-with-confidence-using-visual-regression-testing

**Comment by shimoncohen** [comment]
Need to upgrade preview to use `yarn` instead of `npm`.

File: docusaurus.config.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
 // @ts-check
 // Note: type annotations allow type checking and IDEs autocompletion
 
-const lightCodeTheme = require('prism-react-renderer/themes/github');
-const darkCodeTheme = require('prism-react-renderer/themes/dracula');
+import {themes as prismThemes} from 'prism-react-renderer';
+import type { Config, ReportingSeverity } from '@docusaurus/types';
+import type * as Preset from '@docusaurus/preset-classic';
```
Comment (netanelC) [inline docusaurus.config.ts:6]: Consider importing only the necessary types
```suggestion
import type { Option, ThemeConfig } from '@docusaurus/preset-classic';
```

Comment (netanelC) [inline docusaurus.config.ts:6]: Consider importing only the necessary types
```suggestion
import type { Option, ThemeConfig } from '@docusaurus/preset-classic';
```

**Review by netanelC — APPROVED** [review]

**Pull request merged**